File: packages/expo-font/android/src/main/java/expo/modules/font/FontLoaderRecords.kt
package expo.modules.font

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record
import expo.modules.kotlin.types.OptimizedRecord

@OptimizedRecord
data class FontFaceRecord(
  @Field
  val localUri: String = "",

  @Field
  val weight: Int? = null,

  @Field
  val style: String? = null
) : Record

File: packages/expo-font/ios/FontLoaderRecords.swift
import ExpoModulesCore

// `weight`/`style` are the JS-declared values, used only as a fallback when the font file's
// own traits can't be read.
struct FontFaceRecord: Record {
  @Field var localUri: URL?
  @Field var weight: Int?
  // "normal" | "italic"
  @Field var style: String?
}
